web/src/app/monitors/monitor-create-edit/monitor-create-edit.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { v4 as uuid } from 'uuid';
+
+// Third party modules
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+// Dashboard hub models and services
+import { MonitorService, ProjectService } from '../../core/services/index.service';
+import { MonitorModel, ProjectModel } from '../../shared/models/index.model';
+
+@Component({
+  selector: 'dashboard-monitor-create-edit',
+  templateUrl: './monitor-create-edit.component.html',
+  styleUrls: ['./monitor-create-edit.component.scss'],
+})
+export class MonitorCreateEditComponent implements OnInit {
+
+  private id: string;
+  private monitorsList: MonitorModel[] = [];
+  private projectSubscription: Subscription;
+  private uid: string;
+  public isEdit: Boolean = false;
+  public monitorForm: FormGroup;
+
+  constructor(
+    private form: FormBuilder,
+    private monitorService: MonitorService,
+    private projectService: ProjectService,
+    private route: ActivatedRoute
+  ) { }
+
+  /**
+   * Lifecycle init method
+   */
+  ngOnInit(): void {
+    this.uid = this.route.snapshot.paramMap.get('uid');
+    this.intializeMonitorForm();
+    this.projectSubscription = this.projectService
+      .findOneById(this.uid)
+      .subscribe((project: ProjectModel) => {
+        this.monitorsList = project.monitors ? project.monitors : [];
+        this.id = this.route.snapshot.paramMap.get('id');
```

Comment:
I think we should always use `uid` because they are both the same format. Can use `projectUid` and `monitorUid` to be clearer

Reply:
done

Reply:
done